internal/extproc/server_test.go

```diff
@@ -259,3 +260,55 @@ func TestServer_Process(t *testing.T) {
 		require.Error(t, err, "context canceled")
 	})
 }
+
+func TestFilterSensitiveHeaders(t *testing.T) {
+	logger, buf := newTestLoggerWithBuffer()
+	hm := &corev3.HeaderMap{Headers: []*corev3.HeaderValue{{Key: "foo", Value: "bar"}, {Key: "authorization", Value: "sensitive"}}}
+	filtered := filterSensitiveHeaders(hm, logger, []string{"authorization"})
+	require.Len(t, filtered.Headers, 2)
+	for _, h := range filtered.Headers {
+		if h.Key == "authorization" {
+			require.Equal(t, "[REDACTED]", h.Value)
+		} else {
+			require.Equal(t, "bar", h.Value)
+		}
+	}
+	require.Eventually(t, func() bool {
+		return strings.Contains(buf.String(), "filtering sensitive header")
+	}, 1*time.Second, 100*time.Millisecond, buf.String())
```

Comment:
why do you need eventually?

Reply:
I don't, logging is synchronous and the log message should be written immediately to the buffer. Thanks!